tests/backends/tensorflow/test_gradient.py

```diff
@@ -153,6 +153,50 @@ def test_Squeezing_mean_photon_number_gradient_1_mode():
     assert np.isclose(gradient, 2 * np.sinh(r) * np.cosh(r))
 
 
+def test_CubicPhase_mean_photon_number_gradient_1_mode():
+    gamma = tf.Variable(0.2)
+
+    simulator = pq.TensorflowPureFockSimulator(d=1, config=pq.Config(cutoff=6))
+
+    with tf.GradientTape() as tape:
+        with pq.Program() as program:
+            pq.Q() | pq.Vacuum()
+
+            pq.Q(0) | pq.CubicPhase(gamma=gamma)
+
+        state = simulator.execute(program).state
+
+        mean = state.mean_photon_number()
+
+    gradient = tape.gradient(mean, [gamma])
+
+    assert np.isclose(mean, 0.029567)
+    assert np.isclose(gradient, 0.291416)
```

Comment:
I think for one mode these could be easily calculated explicitely from the evolution of the ladder operators.